scripts/synapse_port_db

```diff
@@ -96,32 +100,10 @@ APPEND_ONLY_TABLES = [
 end_error_exec_info = None
 
 
-class Store(object):
-    """This object is used to pull out some of the convenience API from the
-    Storage layer.
-
-    *All* database interactions should go through this object.
-    """
-
-    def __init__(self, db_pool, engine):
-        self.db_pool = db_pool
-        self.database_engine = engine
-
-    _simple_insert_txn = SQLBaseStore.__dict__["_simple_insert_txn"]
-    _simple_insert = SQLBaseStore.__dict__["_simple_insert"]
-
-    _simple_select_onecol_txn = SQLBaseStore.__dict__["_simple_select_onecol_txn"]
-    _simple_select_onecol = SQLBaseStore.__dict__["_simple_select_onecol"]
-    _simple_select_one = SQLBaseStore.__dict__["_simple_select_one"]
-    _simple_select_one_txn = SQLBaseStore.__dict__["_simple_select_one_txn"]
-    _simple_select_one_onecol = SQLBaseStore.__dict__["_simple_select_one_onecol"]
-    _simple_select_one_onecol_txn = SQLBaseStore.__dict__[
-        "_simple_select_one_onecol_txn"
-    ]
-
-    _simple_update_one = SQLBaseStore.__dict__["_simple_update_one"]
-    _simple_update_one_txn = SQLBaseStore.__dict__["_simple_update_one_txn"]
-    _simple_update_txn = SQLBaseStore.__dict__["_simple_update_txn"]
+class Store(DataStore):
```

Comment:
I'm assuming we're inheriting from DataStore here so that we can get the background-update handlers. However, I'm concerned that instantiating a DataStore also means that we start a bunch of background processes which we really don't want: as a simple example, `censor_redactions` seems like a dangerous thing to run. Have you considered instead pulling out the calls to `register_background_update_handler` to separate classes (or even to standalone functions, or something)?

Reply:
(this might also help with not instantiating an entire HomeServer object)

Reply:
> Have you considered instead pulling out the calls to `register_background_update_handler` to separate classes (or even to standalone functions, or something)?

Yes, but this is not trivial to do, as some classes' background updates have ties within the rest of the class, it's not always possible.
Plus pulling in the whole datastore had the advandage of making it so people don't have to think about adding new classes to this store's list of inherited classes when they add a background job to a class that previously had none, but that doesn't happen a lot and should be tackled by CI in the future.
I'll have another try at doing that.

Reply:
Done (see all commits named `Move xxxxxxxxxxx's bg updates to a dedicated store`)